src/tt.h

```diff
@@ -24,36 +24,57 @@
 #include "misc.h"
 #include "types.h"
 
-/// TTEntry struct is the 10 bytes transposition table entry, defined as below:
+/// TTEntry struct is the 8 bytes transposition table entry, defined as below:
 ///
-/// key        16 bit
-/// move       16 bit
+/// move       13 bit
+/// reserved    3 bit
 /// value      16 bit
 /// eval value 16 bit
 /// generation  5 bit
 /// pv node     1 bit
 /// bound type  2 bit
 /// depth       8 bit
 
+struct TTEntryPacked {
+
+  using Move13     = BitFieldDesc< 0U, 13U, uint16_t>;
+  using ExtraHash3 = BitFieldDesc<13U,  3U, uint8_t>;
+  using Value16    = BitFieldDesc<16U, 16U, Value, true>; // sign-extend
+  using Eval16     = BitFieldDesc<32U, 16U, Value, true>; // sign-extend
+  using Gen5       = BitFieldDesc<48U,  5U, uint8_t>;
+  using Pv         = BitFieldDesc<53U,  1U, bool>;
+  using Bound2     = BitFieldDesc<54U,  2U, Bound>;
+  using Depth8     = BitFieldDesc<56U,  8U, uint8_t>;
```

Comment:
this could use some static_assert to check whether the fields are disjoint.

Reply:
Thanks for the reviews! I'll address these one way or another on my next PR iteration